monai/utils/misc.py

```diff
@@ -821,3 +823,29 @@ def check_kwargs_exist_in_class_init(cls, kwargs):
     extra_kwargs = input_kwargs - init_params
 
     return extra_kwargs == set(), extra_kwargs
+
+
+def run_cmd(cmd_list: list[str], **kwargs: Any) -> subprocess.CompletedProcess:
+    """
+    Run a command by using ``subprocess.run`` with capture_output=True and stderr=subprocess.STDOUT
+    so that the raise exception will have that information. The argument `capture_output` can be set explicitly
+    if desired, but will be overriden with the debug status from the variable.
+
+    Args:
+        cmd_list: a list of strings describing the command to run.
+        kwargs: keyword arguments supported by the ``subprocess.run`` method.
+
+    Returns:
+        a CompletedProcess instance after the command completes.
+    """
+    debug = MONAIEnvVars.debug()
+    kwargs["capture_output"] = kwargs.get("capture_output", debug)
+
+    try:
+        return subprocess.run(cmd_list, **kwargs)
+    except subprocess.CalledProcessError as e:
+        if not debug:
+            raise
+        output = str(e.stdout.decode(errors="replace"))
+        errors = str(e.stderr.decode(errors="replace"))
+        raise RuntimeError(f"subprocess call error {e.returncode}: {errors}, {output}.") from e
```

Comment:
it seems there are random test errors from `test_lmdbdataset`

MONAI-premerge/detail/MONAI-premerge/2590/pipeline

```
[2023-06-20T10:09:22.856Z] ======================================================================
[2023-06-20T10:09:22.856Z] ERROR: test_shape_0 (tests.test_lmdbdataset.TestLMDBDataset)
[2023-06-20T10:09:22.856Z] ----------------------------------------------------------------------
[2023-06-20T10:09:22.856Z] Traceback (most recent call last):
[2023-06-20T10:09:22.856Z]   File "/opt/conda/lib/python3.8/site-packages/nibabel/loadsave.py", line 100, in load
[2023-06-20T10:09:22.856Z]     stat_result = os.stat(filename)
[2023-06-20T10:09:22.856Z] FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpen5q6yik/test_image1_new.nii.gz'
[2023-06-20T10:09:22.856Z] 
[2023-06-20T10:09:22.856Z] During handling of the above exception, another exception occurred:
[2023-06-20T10:09:22.856Z] 
[2023-06-20T10:09:22.856Z] Traceback (most recent call last):
[2023-06-20T10:09:22.856Z]   File "/opt/conda/lib/python3.8/site-packages/parameterized/parameterized.py", line 620, in standalone_func
[2023-06-20T10:09:22.856Z]     return func(*(a + p.args), **p.kwargs, **kw)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_lmdbdataset.py", line 203, in test_shape
[2023-06-20T10:09:22.856Z]     dataset_postcached.set_data(data=test_data_new)  # filename list updated, files do not exist
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 605, in set_data
[2023-06-20T10:09:22.856Z]     self._read_env = self._fill_cache_start_reader(show_progress=self.progress)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 631, in _fill_cache_start_reader
[2023-06-20T10:09:22.856Z]     val = self._pre_transform(deepcopy(item))  # keep the original hashed
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 330, in _pre_transform
[2023-06-20T10:09:22.856Z]     item_transformed = self.transform(item_transformed, end=first_random, threading=True)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/compose.py", line 322, in __call__
[2023-06-20T10:09:22.856Z]     result = execute_compose(
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/compose.py", line 111, in execute_compose
[2023-06-20T10:09:22.856Z]     data = apply_transform(
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/transform.py", line 141, in apply_transform
[2023-06-20T10:09:22.856Z]     return _apply_transform(transform, data, unpack_items, lazy, overrides, log_stats)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/transform.py", line 98, in _apply_transform
[2023-06-20T10:09:22.856Z]     return transform(data, lazy=lazy) if isinstance(transform, LazyTrait) else transform(data)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/io/dictionary.py", line 164, in __call__
[2023-06-20T10:09:22.856Z]     data = self._loader(d[key], reader)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/io/array.py", line 263, in __call__
[2023-06-20T10:09:22.856Z]     img = reader.read(filename)
[2023-06-20T10:09:22.856Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/image_reader.py", line 907, in read
[2023-06-20T10:09:22.856Z]     img = nib.load(name, **kwargs_)
[2023-06-20T10:09:22.856Z]   File "/opt/conda/lib/python3.8/site-packages/nibabel/loadsave.py", line 102, in load
[2023-06-20T10:09:22.856Z]     raise FileNotFoundError(f"No such file or no access: '{filename}'")
[2023-06-20T10:09:22.856Z] FileNotFoundError: No such file or no access: '/tmp/tmpen5q6yik/test_image1_new.nii.gz'
```

Reply:
I met it here, re-ran the test, and found it was gone:
https://github.com/Project-MONAI/MONAI/actions/runs/5313380724/jobs/9624309763

Did you see it elsewhere @wyli ?

Reply:
I thought it was an issue with the return code, but it works fine.. I'll try to merge this, and debug further if needed.

Reply:
the issue might be that the system flag is modified during the unit test, I try to keep the original value https://github.com/Project-MONAI/MONAI/pull/6596/commits/dee12cc5cc618a3598ea1cadec0b1f4434c4a4a1

Reply:
Thanks @wyli . I thought the system flag would be no longer valid after the `tests/test_monai_utils_misc.py` finishes. And yes, now this way is better